File: Classes/Logger.cs
using System;
using System.IO;

namespace ADUserLibrary
{
    public enum LogLevel
    {
        Info = 1,
        Warning = 2,
        Error = 3
    }

    public class Logger
    {
        public string LogFilePath { get; set; }
        public LogLevel MinimumLogLevel { get; set; }

        public Logger(string logFilePath = ".\\log.txt", LogLevel minimumLogLevel = LogLevel.Info)
        {
            LogFilePath = logFilePath;
            MinimumLogLevel = minimumLogLevel;
        }

        private void WriteLogEntry(LogLevel level, string message)
        {
            if ((int)level >= (int)MinimumLogLevel)
            {
                string logEntry = $"{DateTime.Now:yyyy-MM-dd HH:mm:ss} [{level}] {message}";
                File.AppendAllText(LogFilePath, logEntry + Environment.NewLine);
                Console.WriteLine(logEntry);
            }
        }

        public void Info(string message)
        {
            WriteLogEntry(LogLevel.Info, message);
        }

        public void Warning(string message)
        {
            WriteLogEntry(LogLevel.Warning, message);
        }

        public void Error(string message)
        {
            WriteLogEntry(LogLevel.Error, message);
        }
    }
}

File: Classes/Logger.java
package com.example.aduser;

import java.io.FileWriter;
import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Logger {
    public enum LogLevel {
        INFO,
        WARNING,
        ERROR
    }

    private String logFilePath;
    private LogLevel minimumLogLevel;

    public Logger(String logFilePath, LogLevel minimumLogLevel) {
        this.logFilePath = logFilePath;
        this.minimumLogLevel = minimumLogLevel;
    }

    private void writeLogEntry(LogLevel level, String message) {
        if (level.ordinal() >= minimumLogLevel.ordinal()) {
            String logEntry = String.format("%s [%s] %s",
                    LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")),
                    level, message);
            try (FileWriter writer = new FileWriter(logFilePath, true)) {
                writer.write(logEntry + System.lineSeparator());
            } catch (IOException e) {
                e.printStackTrace();
            }
            System.out.println(logEntry);
        }
    }

    public void info(String message) {
        writeLogEntry(LogLevel.INFO, message);
    }

    public void warning(String message) {
        writeLogEntry(LogLevel.WARNING, message);
    }

    public void error(String message) {
        writeLogEntry(LogLevel.ERROR, message);
    }
}